app/api/user_endpoint.py

```diff
@@ -36,23 +36,29 @@ async def set_user_preference(key: str, preference: Union[dict, str, int, float]
     """
     Sets user preference.Uses key to set the preference
     """
-    pass
+    user_preference = storage.driver.user.preference[key]
+    if user_preference:
+        user_preference.update({key: preference})
```

Comment:
update user's preferences in User object that you get from args and then save it